libsolidity/formal/SMTEncoder.h

```diff
@@ -238,11 +238,12 @@ class SMTEncoder: public ASTConstVisitor
 	void visitFunctionIdentifier(Identifier const& _identifier);
 	virtual void visitPublicGetter(FunctionCall const& _funCall);
 
-	/// @returns true if @param _contract is set for analysis in the settings
-	/// and it is not abstract.
-	bool shouldAnalyze(ContractDefinition const& _contract) const;
-	/// @returns true if @param _source is set for analysis in the settings.
-	bool shouldAnalyze(SourceUnit const& _source) const;
+	/// @returns true if symbolic representation of @param _contract is required for verification
+	bool shouldEncode(ContractDefinition const& _contract) const;
+	/// @returns true if the verification targets of @param _contract are actually selected for verification
+	bool shouldAnalyzeVerificationTargetsFor(ContractDefinition const& _contract) const;
+	/// @returns true if the contracts in @param _source should be verified
```

Comment:
Would be good to clarify that this is not simply equivalent to checking whether this is true for any of the contracts. It seems possible for this to return `true` even though there are no contracts we should verify.

Maybe like this?
```suggestion
	/// @returns true if we should descend into @param _source to look for contracts that should be verified
```

Reply:
I used your wording.